src/modules/navigator/rtl.cpp

```diff
@@ -309,20 +309,28 @@ RTL::advance_rtl()
 	case RTL_STATE_RETURN:
 		_rtl_state = RTL_STATE_DESCEND;
 
-		if (_navigator->get_vstatus()->is_vtol && !_navigator->get_vstatus()->is_rotary_wing) {
-			_rtl_state = RTL_STATE_TRANSITION_TO_MC;
-		}
-
 		break;
 
 	case RTL_STATE_TRANSITION_TO_MC:
-		_rtl_state = RTL_STATE_RETURN;
+
+		// Only go to land if autoland is enabled.
+		if (_param_land_delay.get() < -DELAY_SIGMA || _param_land_delay.get() > DELAY_SIGMA) {
+			_rtl_state = RTL_STATE_LOITER;
+
+		} else {
+			_rtl_state = RTL_STATE_LAND;
+		}
+
 		break;
 
 	case RTL_STATE_DESCEND:
 
-		// Only go to land if autoland is enabled.
-		if (_param_land_delay.get() < -DELAY_SIGMA || _param_land_delay.get() > DELAY_SIGMA) {
+		// if we are a VTOL and in fixed wing mode then we should now transition to rotary wing mode
+		if (_navigator->get_vstatus()->is_vtol && !_navigator->get_vstatus()->is_rotary_wing) {
+			_rtl_state = RTL_STATE_TRANSITION_TO_MC;
```

Comment:
Can you reorder the switch so it follows the intended order of execution.

`RTL_STATE_RETURN -> RTL_STATE_DESCEND -> RTL_STATE_TRANSITION_TO_MC -> ...`